external/fv3fit/fv3fit/keras/_models/shared/training_loop.py

```diff
@@ -0,0 +1,81 @@
+import tensorflow as tf
+from typing import Iterable, Optional, Sequence, Union, Mapping, Tuple, Callable
+import dataclasses
+import numpy as np
+from fv3fit.keras._models._sequences import _ThreadedSequencePreLoader
+from loaders.batches import shuffle
+import logging
+
+
+# Description of the training loss progression over epochs
+# Outer array indexes epoch, inner array indexes batch (if applicable)
+EpochLossHistory = Sequence[Sequence[Union[float, int]]]
+History = Mapping[str, EpochLossHistory]
+
+logger = logging.getLogger(__file__)
+
+
+@dataclasses.dataclass
+class EpochResult:
+    """
+    Attributes:
+        epoch: count of epoch (starts at zero)
+        history: return value of `model.fit` from each batch
+    """
+
+    epoch: int
+    history: Sequence[tf.keras.callbacks.History]
+
+
+@dataclasses.dataclass
+class TrainingLoopConfig:
+    """
+    epochs: number of times to run through the batches when training
+    workers: number of workers for parallelized loading of batches fed into
+        training, if 1 uses serial loading instead
+    max_queue_size: max number of batches to hold in the parallel loading queue
+    keras_batch_size: actual batch_size to pass to keras model.fit,
+        independent of number of samples in each data batch in batches
+    """
+
+    epochs: int = 3
+    workers: int = 1
+    max_queue_size: int = 8
+    keras_batch_size: int = 16
```

Comment:
`DenseModel.fit_kwargs` used to call this `batch_size` while the earlier PrecipitativeModel and this config call it `keras_batch_size`... IMO now that it's in this config it is more clear that it's the "keras" batch size so the prefix is redundant and it'd be less error prone for people converting the old flatter hyperparameters to the newer version if it is kept as `batch_size`.

Reply:
Renamed it to `batch_size`.